### DISCUSSION TITLE
Creating contextMenu in Avalonia

### DISCUSSION THREAD

**Discussion by santhoshkumarmk** [AvaloniaUI/Avalonia#6497] [General]
Hi,

How to create contextmenu from backend code in avalonia as like wpf

**Comment by maxkatz6** [comment]
```csharp
button.ContextFlyout = new MenuFlyout
{
    Items = new [] 
   {
       new MenuItem { Header = "First Item" }
   }
}
```

**Reply by maxkatz6** [reply]
Or legacy WPF ContextMenu also works:
```
button.ContextMenu = new ContextMenu
{
    Items = new [] 
   {
       new MenuItem { Header = "First Item" }
   }
}
```

**Reply by santhoshkumarmk** [reply]
when i run this code menu is not populating 

 Button button = new Button();
            button.ContextMenu = new ContextMenu
            {
                Items = new[]
   {
       new MenuItem { Header = "First Item" }
   }
            };

**Reply by maxkatz6** [reply]
Well, you are doing something wrong

```
public MainWindow()
{
    this.InitializeComponent();
    this.AttachDevTools();

    Button button = new Button();
    button.Content = "ASDF";
    button.ContextMenu = new ContextMenu
    {
        Items = new[]
        {
            new MenuItem { Header = "First Item" }
        }
    };
    this.Content = button;
}
```

![image](https://user-images.githubusercontent.com/3163374/131440226-6a43ac11-35b3-4eb7-9f73-dde2c01c58bb.png)

**Comment by santhoshkumarmk** [comment]
and IsOpen property is false by default . when i try to set externally, i am unable to do
because this is protected .


button.ContextMenu.IsOpen = true;

**Reply by maxkatz6** [reply]
You shouldn't set it to true

**Reply by maxkatz6** [reply]
And you shouldn't try to open context menu manually. It's a **context** menu that is expected to be invoked on right click.

**Reply by maxkatz6** [reply]
If you want to show a popup or flyout on any action, you can either create Popup and set its IsOpen to true (in the same way as in WPF), 
Or it's better to use flyout
```
Button button = new Button();
button.Content = "ASDF";
Content = button;

button.Click += (s, a) =>
{
    MenuFlyout flyout = new MenuFlyout()
    {
        Items = new[]
        {
            new MenuItem { Header = "First Item" }
        },
        Placement = FlyoutPlacementMode.Bottom
    };
    FlyoutBase.SetAttachedFlyout(button, flyout);
    FlyoutBase.ShowAttachedFlyout(button);
    // Or you can just call 'flyout.ShowAt(button);' without 'AttachedFlyout'. The later one only for convenience, but not required.
};
```

**Reply by santhoshkumarmk** [reply]
Hi Max,

I called this function on button click event when i clicked the whole content is replaced with this button in UI.

I don't want this behavior just want in a pop up as like WPF

**Comment by santhoshkumarmk** [comment]
Its working when i do right click . But the problem is here the context menu occupying whole screen . I am not
able view my prvious UI controls
